doc/proposals/ansible-operator.md

```diff
@@ -0,0 +1,101 @@
+
+### New Operator Type
+
+This proposal creates a new type of operator called `ansible`.  The new type is used to tell the tooling to act on that type of operator. 
+
+### Package Structure
+Packages will be added to the operator-sdk. These packages are designed to be usable by the end user if they choose to and should have a well documented public API. The proposed packages are:
+* /operator-sdk/pkg/ansible/handler
+  * Will contain a new Handler Interface to allow users to define how to handle ansible events.
+  * Will contain a wrapper type that conforms to the sdk.Handler, for a user to use with standard operator-sdk workflows. The wrapper type has default methods for everything, but they are overridable.
+* /operator-sdk/pkg/ansible/runner
+  * Contains the runner types and interfaces
+  * Implementation is behind an internal package (/operator-sdk/pkg/ansible/runner/internal)
+  * New Methods are exposed and are the main way a user interacts with the package
+  * Runner interface for running ansible from the operator.
+  * NewForWatchers - the method that returns a map of GVK to Runner types based on the watchers file.
+  * NewPlaybookRunner - the method that returns a new Runner for a playbook.
+  * NewRoleRunner - the method that returns a new Runner for a role.
+  * This contains the events API code and public methods. Implementation should probably be in the internal package. The events API is used for recieving events from ansible runner. 
+
+* /operator-sdk/pkg/ansible/proxy
+  * This is a reverse proxy for the kubernetes API that is used for owner reference injection.
+* /operator-sdk/pkg/ansible/proxy/kubeconfig
+  * Code needed to generate the kubeconfig for the proxy.
+
+
+### Commands
+We are adding and updating existing commands to accommodate the ansible operator.  Changes to the `cmd` package as well as changes to the generator are needed. 
+
+`operator-sdk new --type ansible --kind <kind> --api-version <group/version>`  This will be a new generation command under the hood. We will:
+* Create a new ansible role in the roles directory
+* Create a new watchers file. The role and GVK are defaulted based on input to the command. 
+* A CRD is generated. This CRD does not have any validations defined. 
+* A dockerfile is created using the watchers file and the ansible role with the base image being the ansible operator base image.
+
+The resulting structure should be
+```
+|- Dockerfile
+|
+|- roles
+\ | - <kind>
+|  \ | - generated ansible role
+|
+| - watchers.yaml
+|
+| - deploy
+\  | - <kind>-CRD.yaml
+|  | - rbac.yaml
+|  | - operator.yaml
+|  | - cr.yaml
+```
+
+`operator-sdk generate crd <api-version> <kind> ` This will be used to generate new CRDs based on ansible code. The command helps when a user wants to watch more than one CRD for their operator.
```

Comment:
We probably want to have this functionality for the go operator as well. I think once we switch to controller runtime, we can do that for the go operator. The question I am having now is that how do we generate additional CRDs for both go type and ansible type?

Reply:
`while this command (operator-sdk generate crd <api-version> <kind>) only generates the CRD.`

as long as it can generate a CRD regardless whether the operator is a Go one or Ansible one. I think this command makes sense.